### PR TITLE
Add support for AD7768, and by extension, CN0501

### PR DESCRIPTION
**Pull request by mthoren-adi**
dma_tests.py: change np.sum to np.max to prevent rollover

Signed-off-by: Mark Thoren <mark.thoren@analog.com>

# Description

Add support for AD7768, 8-channel simultaneous sampling sigma-delta ADC.
dma_tests.py updated - using np.sum causes rollover to negative number in some circumstances.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How has this been tested?
**Test Configuration**:

* Hardware:
1: AD7768 Eval Board on Zed, CN0501
2: CN0501 on Cora Z7
Boot files for both are present on the 2021_r1/RC2 Kuiper image.
* OS: Windows 10
A pytest fixture tests DMA capture, as well as all writeable attributes (sample rate, filter type, power mode.)

# Documentation

All doc strings added to supported_parts.md, index.rst

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [XI have commented my code, particularly in hard-to-understand areas
- [X] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mthoren-adi** [comment]
@scuciurean, @tfcollins , most of this should be pretty straightforward. Please take a look at the channel initialization though:
[https://github.com/analogdevicesinc/pyadi-iio/pull/344/files#diff-432f14712e50c6d2efb6330cba77efcd043bf2f016409556712ec5d88773c579R64](url)
I think "probing" for channels (versus explicitly listing channel names) is probably a good practice for most multichannel parts, particularly if the number / name of channels can be changed via the device tree. In this case - I wanted to be prepared to add support for the AD7768-4, which is nearly identical but with only 4 channels.

**Comment by scuciurean** [comment]
> 

Ad7768-1/-4 are using https://wiki.analog.com/resources/fpga/docs/util_cpack 

> @scuciurean, @tfcollins , most of this should be pretty straightforward. Please take a look at the channel initialization though: [https://github.com/analogdevicesinc/pyadi-iio/pull/344/files#diff-432f14712e50c6d2efb6330cba77efcd043bf2f016409556712ec5d88773c579R64](url) I think "probing" for channels (versus explicitly listing channel names) is probably a good practice for most multichannel parts, particularly if the number / name of channels can be changed via the device tree. In this case - I wanted to be prepared to add support for the AD7768-4, which is nearly identical but with only 4 channels.

File: adi/ad7768.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+        self._ctrl = self._ctx.find_device("ad7768")
+        self._rxadc = self._ctx.find_device("ad7768")
+        self._device_name = "ad7768"
+        if not self._rxadc:
+            self._ctrl = self._ctx.find_device("cf_axi_adc")
+            self._rxadc = self._ctx.find_device("cf_axi_adc")
+            self._device_name = "cf_axi_adc"
+
+        rx._rx_stack_interleaved = True  # DMA config. dependent. True for ad7768.
```
Comment (tfcollins) [inline adi/ad7768.py:63]: Can be removed. This doesn't do anything anymore

File: adi/ad7768.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        rx._rx_data_type = np.int32
```
Comment (tfcollins) [inline adi/ad7768.py:64]: Can you put this in the class definition

File: adi/ad7768.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+# def _cast32(sample):
```
Comment (tfcollins) [inline adi/ad7768.py:38]: Remove commented block

**Review by tfcollins — CHANGES_REQUESTED** [review]

**Comment by mthoren-adi** [comment]
Thanks for the review @tfcollins , should be all set. Did a quick full-scale reality check to make sure nothing went awry with the data types:

Capture

**Pull request merged**